niimpy/location.py

```diff
@@ -134,24 +134,31 @@ def extract_distance_features(location, column_prefix=None):
     """
     def compute_distance_features(df):
         """Compute features for a single user"""
-        total_dist = 0
-        speeds = []
-        for i in range(df.shape[0] - 2):
-            loc1 = df.iloc[i][['double_latitude', 'double_longitude']]
-            loc2 = df.iloc[i + 1][['double_latitude', 'double_longitude']]
-            dist = geodesic(loc1, loc2).meters
-            total_dist += dist
-
-            time_delta = (df.index[i + 1] - df.index[i]).seconds
-            if time_delta > 0:
-                speeds.append(dist / time_delta)
-        speed_average = np.mean(speeds)
-        speed_variance = np.var(speeds)
+        df = df.sort_index()  # sort based on time
+
+        dists = np.zeros(df.shape[0])
+        time_deltas = np.zeros(df.shape[0])
+
+        for i in range(1, df.shape[0]):
+            loc1 = df.iloc[i - 1][['double_latitude', 'double_longitude']]
```

Comment:
This step is after doing preprocessing and binning, so it is expected that coordinates are sensible. At least in MMM dataset, there is no null latitude or longitude. I'll some `assert` statement at the start of function to make sure that there is no null value.

Reply:
👍 this function would be reused in some other places also, so I think it is good to assert that the input dataframe doesn't contain null values. Otherwise, everything else looks good!